metaflow/plugins/kubernetes/kubernetes_cli.py

```diff
@@ -175,7 +182,13 @@ def echo(msg, stream="stderr", job_id=None):
     stderr_location = ds.get_log_location(TASK_LOG_SOURCE, "stderr")
 
     # `node_selector` is a tuple of strings, convert it to a dictionary
-    node_selector = KubernetesDecorator.parse_node_selector(node_selector)
+    node_selector = KubernetesDecorator.parse_kube_list(node_selector)
+
+    # `labels` is a tuple of strings or a tuple with a single comma separated string
+    # convert it to a dict
+    labels = KubernetesDecorator.parse_kube_list(
```

Comment:
The string generated in the `runtime_step_cli` in the `kubernetes_decorator.py` ends up looking something like:

```
labels = ("label1=val1,label2=val2",)
```

This seems somewhat by design for the `node_selector` field.  Hence there is a bit of additional logic here.

Reply:
https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/kubernetes/kubernetes_decorator.py#L288

I suspect this should not have `",".join(...)`.  I will test if removing that fixes the issue.